src/main/groovy/me/rexim/morganey/gradle/MorganeyPlugin.groovy

```diff
@@ -2,16 +2,55 @@ package me.rexim.morganey.gradle
 
 import org.gradle.api.Plugin
 import org.gradle.api.Project
+import org.gradle.api.plugins.JavaPlugin
 
 /**
  * Created by thoma on 03/10/2016.
  */
-class MorganeyPlugin implements Plugin<Project>{
+class MorganeyPlugin implements Plugin<Project> {
+    ArrayList<File> filesToAdd = new ArrayList<File>()
 
     @Override
     void apply(Project project) {
-        project.task('hello') << {
+        project.plugins.apply(JavaPlugin.class)
+        project.task('hello') << { // This is to be used for testing purposes only
             println "Hello from the Morganey Plugin"
         }
+        project.task('package-module') << {
+            println "Packaging Morganey Module"
+            getFiles(project, project.projectDir.path + "/src")
+            for (int i = 0; i < filesToAdd.size(); i++) {
+                println "Adding File: " + filesToAdd.get(i).name
+            }
+        }
+        project.task('build-jar') << {
+            String[] paths = new String[filesToAdd.size()]
+            for(int i = 0; i < paths.length; i++){
+                println filesToAdd.get(i).absolutePath
+                paths[i] = filesToAdd.get(i).absolutePath
+            }
+                type: 'jar'
```

Comment:
@06needhamt  I'm not an expert in gradle plugins, but why did you think it will just magically create a jar? I tried to google all of that and didn't find anything. Could you please elaborate on why you write it like that?